extensions/ql-vscode/src/databases/db-item-selection.ts

```diff
@@ -42,3 +43,49 @@ function extractSelected(
   }
   return undefined;
 }
+
+export function mapDbItemToSelectedDbItem(
+  dbItem: DbItem,
+): SelectedDbItem | undefined {
+  switch (dbItem.kind) {
+    case DbItemKind.RootLocal:
+    case DbItemKind.RootRemote:
+      // Root items are not selectable.
+      return undefined;
+
+    case DbItemKind.LocalList:
+      return {
+        kind: SelectedDbItemKind.LocalUserDefinedList,
+        listName: dbItem.listName,
+      };
+    case DbItemKind.RemoteUserDefinedList:
+      return {
+        kind: SelectedDbItemKind.RemoteUserDefinedList,
+        listName: dbItem.listName,
+      };
+    case DbItemKind.RemoteSystemDefinedList:
+      return {
+        kind: SelectedDbItemKind.RemoteSystemDefinedList,
+        listName: dbItem.listName,
+      };
+    case DbItemKind.RemoteOwner:
+      return {
+        kind: SelectedDbItemKind.RemoteOwner,
+        ownerName: dbItem.ownerName,
+      };
+
+    case DbItemKind.LocalDatabase:
+      return {
+        kind: SelectedDbItemKind.LocalDatabase,
+        listName: dbItem?.parentListName,
+        databaseName: dbItem.databaseName,
+      };
+
+    case DbItemKind.RemoteRepo:
+      return {
+        kind: SelectedDbItemKind.RemoteRepository,
+        listName: dbItem?.parentListName,
+        repositoryName: dbItem.repoFullName,
+      };
```

Comment:
Very minor (random extra whitespace) 🤷🏽 😄 

```suggestion
      };
    case DbItemKind.LocalDatabase:
      return {
        kind: SelectedDbItemKind.LocalDatabase,
        listName: dbItem?.parentListName,
        databaseName: dbItem.databaseName,
      };
    case DbItemKind.RemoteRepo:
      return {
        kind: SelectedDbItemKind.RemoteRepository,
        listName: dbItem?.parentListName,
        repositoryName: dbItem.repoFullName,
      };
```